WooCommerce/WooCommerceTests/ViewRelated/Orders/Simple Payments/PriceFieldFormatterTests.swift

```diff
@@ -114,29 +113,85 @@ final class PriceFieldFormatterTests: XCTestCase {
         XCTAssertEqual(priceFieldFormatter.formattedAmount, "€0,00")
     }
 
-    func test_formatter_works_with_negative_numbers() {
+    func test_formatUserInput_works_with_negative_numbers() {
         // Given
         let priceFieldFormatter = PriceFieldFormatter(locale: usLocale, storeCurrencySettings: usStoreSettings, allowNegativeNumber: true)
 
         // When & Then
-        _ = priceFieldFormatter.formatAmount("-12")
+        _ = priceFieldFormatter.formatUserInput("-12")
         XCTAssertEqual(priceFieldFormatter.formattedAmount, "-$12")
 
         // When & Then
-        _ = priceFieldFormatter.formatAmount("-hi:11.3030-")
+        _ = priceFieldFormatter.formatUserInput("-hi:11.3030-")
         XCTAssertEqual(priceFieldFormatter.formattedAmount, "-$11.30")
     }
 
-    func test_formatter_disallows_negative_numbers_by_default() {
+    func test_formatUserInput_disallows_negative_numbers_by_default() {
         // Given
         let priceFieldFormatter = PriceFieldFormatter(locale: usLocale, storeCurrencySettings: usStoreSettings)
 
         // When & Then
-        _ = priceFieldFormatter.formatAmount("-12")
+        _ = priceFieldFormatter.formatUserInput("-12")
         XCTAssertEqual(priceFieldFormatter.formattedAmount, "$12")
 
         // When & Then
-        _ = priceFieldFormatter.formatAmount("-hi:11.3030-")
+        _ = priceFieldFormatter.formatUserInput("-hi:11.3030-")
         XCTAssertEqual(priceFieldFormatter.formattedAmount, "$11.30")
     }
+
+    func test_formatUserInput_supports_numbers_with_decimal_and_thousands_separator() {
+        // Given
+        let priceFieldFormatter = PriceFieldFormatter(locale: usLocale, storeCurrencySettings: usStoreSettings)
+
+        // When
+        _ = priceFieldFormatter.formatUserInput("1,000.40")
+
+        // Then
+        XCTAssertEqual(priceFieldFormatter.formattedAmount, "$1000.40")
+    }
+
+    func test_formatUserInput_upports_numbers_with_decimal_and_thousands_separator_uses_custom_separators() {
+        // Given
+        let customSettings = CurrencySettings(currencyCode: .EUR, currencyPosition: .left, thousandSeparator: ".", decimalSeparator: ",", numberOfDecimals: 2)
+        let priceFieldFormatter = PriceFieldFormatter(locale: Locale(identifier: "lt-LT"), storeCurrencySettings: customSettings)
+
+        // When
+        _ = priceFieldFormatter.formatUserInput("1000,5")
+
+        // Then
+        XCTAssertEqual(priceFieldFormatter.formattedAmount, "€1000,5")
```

Comment:
I was curious to see if this fails if we assert against `€1000,50` since we pass `numberOfDecimals: 2` and it fails (expected since we're not validating this after the fact), but maybe we want to add an additional test to check if cuts down to the expected number of capped decimals if the user input happens to be more:

```
    func test_formatUserInput_when_decimals_more_than_expected_then_returns_the_max_numberOfDecimals_allowed() {
        // Given
        let customSettings = CurrencySettings(currencyCode: .EUR,
                                              currencyPosition: .left,
                                              thousandSeparator: ".",
                                              decimalSeparator: ",",
                                              numberOfDecimals: 2)

        let priceFieldFormatter = PriceFieldFormatter(locale: Locale(identifier: "lt-LT"), storeCurrencySettings: customSettings)

        // When
        _ = priceFieldFormatter.formatUserInput("1000,5123")

        // Then
        XCTAssertEqual(priceFieldFormatter.formattedAmount, "€1000,51")
```

Reply:
Thanks for the suggestion, added.